File: Compiler design/Lab4/Java/src/ru/bmstu/iu9/Message.java
package ru.bmstu.iu9;

public class Message {
    public final boolean isError;
    public final String text;
    public final Position position;

    public Message(boolean isError, String text, Position position) {
        this.isError = isError;
        this.text = text;
        this.position = position;
    }
}

File: Compiler design/Lab4/OCaml/Message.ml
open Batteries;;

class message is_error_ text_ position_ = object (self)
    
    val is_error: bool = is_error_
    val text: BatUTF8.t = text_
    val position: Position.position = position_

    method get_error = is_error
    method get_text = text
    method get_position = position

    end;;
